application/CMakeLists.txt

```diff
@@ -56,11 +56,27 @@ elseif (MSVC)
   set_target_properties(f3d PROPERTIES VS_DEBUGGER_ENVIRONMENT "PATH=$<TARGET_FILE_DIR:VTK::CommonCore>") # Add VTK in MSVC PATH environment variable
 endif ()
 
-target_include_directories(f3d
-  PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/external>
+if (F3D_USE_EXTERNAL_CXXOPTS)
+  find_package(cxxopts REQUIRED)
+  target_link_libraries(f3d PRIVATE cxxopts::cxxopts)
+endif ()
+if (F3D_USE_EXTERNAL_NLOHMANN_JSON)
+  find_package(nlohmann_json REQUIRED)
+  target_link_libraries(f3d PRIVATE nlohmann_json::nlohmann_json)
+endif ()
+
+if (F3D_USE_EXTERNAL_CXXOPTS AND F3D_USE_EXTERNAL_NLOHMANN_JSON)
+  target_include_directories(f3d
+    PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
   )
+else ()
+  target_include_directories(f3d
```

Comment:
you can do multiple target_include_directories calls I think

Reply:
I do not get your point here. Could you say more expliciltly?

Reply:
you could write it like this, if you put the headers in their own dir

```
target_include_directories(f3d PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
if (F3D_USE_EXTERNAL_CXXOPTS)
  find_package(cxxopts REQUIRED)
  target_link_libraries(f3d PRIVATE cxxopts::cxxopts)
else ()
  target_include_directories(f3d PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/external/cxxopts>
endif ()

if (F3D_USE_EXTERNAL_NLOHMANN_JSON)
  find_package(nlohmann_json REQUIRED)
  target_link_libraries(f3d PRIVATE nlohmann_json::nlohmann_json)
else ()
  target_include_directories(f3d PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/external/json>
endif ()
```

This way we would be sure of which include is done, which is not the case currently